OpenSSL/crypto.py

```diff
@@ -1340,6 +1340,19 @@ def get_extension(self, index):
         extension = _lib.X509_EXTENSION_dup(ext._extension)
         ext._extension = _ffi.gc(extension, _lib.X509_EXTENSION_free)
         return ext
+        
+    def del_extension(self, index):
+        """
+        Delete a specific extension of the certificate by index.
+
+        :param index: The index of the extension to delete.
+        :return: The X509Extension object deleted at the specified index.
+        """
+        ext = X509Extension.__new__(X509Extension)
+        ext._extension = _lib.X509_delete_ext(self._x509, index)
```

Comment:
This will leak memory. It needs to use `X509_EXTENSION_free` in conjunction with `ffi.gc`. Take a look at the method right above this for an example.